csharp/src/Ice/OutgoingAsync.cs

```diff
@@ -541,6 +659,25 @@ protected void InvokeImpl(bool userThread)
                             Observer.Retried();
                         }
                     }
+                    catch (Ice.RemoteException ex)
+                    {
+                        if (ChildObserver != null)
+                        {
+                            ChildObserver.Failed(ex.TypeId);
+                            ChildObserver.Detach();
+                            ChildObserver = null;
+                        }
+                        int interval = Proxy.IceHandleException(ex, Handler, IsIdempotent, _sent, ref _cnt);
+                        if (interval > 0)
+                        {
+                            Communicator.AddRetryTask(this, interval);
+                            return;
+                        }
+                        else if (Observer != null)
+                        {
+                            Observer.Retried();
+                        }
+                    }
```

Comment:
I don't think this is need, the code from the try block doesn't handle the response from the server.

Reply:
Fixing.